generator/send_receive.py

```diff
@@ -0,0 +1,54 @@
+import sys
+sys.path.append("ParseCAN")
+import ParseCAN
+from common import send_recieve_path, coord, templ, ifndef, endif, frame_handler
+
+
+def define_pub_frame(frame, name_prepends, busnm, fw):
+    tot_name = coord(name_prepends, frame.name,
+        prefix=False)
+    fw('void send_{}_msg(CANlib_{}_T *inp)'.format(
+        tot_name, tot_name) + ' {\n'
+    )
+
+    if frame.period is not None:
+        fw('\tLIMIT(CANlib_{});\n'.format(tot_name))
+
+    fw(
+        '\tFrame frame;\n' +
+        '\tCANlib_Pack_' + tot_name + '(inp, &frame);\n' +
+        '\tCANlib_TransmitFrame(&frame,' + busnm + ');\n'
+        '}\n\n'
+    )
+
+
+def define_sub_frame(frame, name_prepends, fw):
+    tot_name = coord(name_prepends, frame.name,
+        prefix=False)
+    fw('void handle_{}_msg(Frame *frame)'.format(
```

Comment:
Can this be weakly linked so that people may define their own on the interrupt?

Reply:
Also consistent naming for this too pls.

Reply:
I disagree because then you'd necessarily have to have an unnecessary struct updated for you and you'll also have to do the switch case on your own again which is what this handle is meant to avoid.

Reply:
Actually can you explain why weakly linking it will fix this problem? If they need access to the switch case that's inside the update function, in which case they're out of luck...
I also think we can just wait until we actually have interrupted CAN to support it in the lib